File: python/scripts/schunk_plt.py
import pandas as pd
import matplotlib.pyplot as plt

df = pd.read_csv("../../data/admittance_control/test_1_020326/maze_damp_80.000000_01.csv")

# Fix timestamp: replace last ':' with '.'
df["TimeStamp "] = df["TimeStamp "].str.replace(
    r"(\d{2}:\d{2}:\d{2}):", r"\1.", regex=True
)

# Convert to datetime
df["TimeStamp "] = pd.to_datetime(df["TimeStamp "], format="%H:%M:%S.%f")

# Convert to relative time (seconds)
t0 = df["TimeStamp "].iloc[0]
df["t"] = (df["TimeStamp "] - t0).dt.total_seconds()

plt.figure()
plt.plot(df["t"], df[" Q1"], label="Q1")
# plt.plot(df["t"], df["Q2"], label="Q2")
# plt.plot(df["t"], df["Q3"], label="Q3")
plt.xlabel("Time [s]")
plt.ylabel("Joint position [rad]")
plt.legend()
plt.grid(True)
plt.show()

File: matlab/schunk_plt.m
clc; clear;

csvFile = "../data/admittance_control/test_1_020326/maze_damp_80.000000_01.csv";  % TODO

opts = detectImportOptions(csvFile, 'Delimiter', ',');
opts.VariableNamingRule = "preserve";   % keep original header text
tsCol = opts.VariableNames{contains(opts.VariableNames, "TimeStamp")};
opts = setvartype(opts, tsCol, "string");
opts = setvaropts(opts, tsCol, 'WhitespaceRule', 'trim');

T = readtable(csvFile, opts);
t = datetime(T.TimeStamp, 'InputFormat', 'HH:mm:ss:SSSSSS');
% t = seconds(t - t(1));
Q  = T{:, {'Q1','Q2','Q3','Q4','Q5','Q6'}};
dQ = T{:, {'dQ1','dQ2','dQ3','dQ4','dQ5','dQ6'}};
FT = T{:, {'FT1','FT2','FT3','FT4','FT5','FT6'}};

figure; hold on; grid on;

% subplot(3,1,1); hold on;
% plot(t, Q(:,1)); title('Joint Positions'); ylabel('rad');

% subplot(3,1,2); hold on;
% plot(t, dQ); title('Joint Velocities'); ylabel('rad/s');

% subplot(3,1,3); hold on;
plot(t, FT(:,1:2)); title('Force/Torque'); ylabel('N / Nm'); xlabel('Time');
